### ISSUE TITLE
The evaluations

### ISSUE THREAD

**Issue by cleverer123** [Mehran-k/SimplE#1]
I really appreciate your work but I'm not sure about one thing: In `Reader.py next_batch()`, if the `neg_ratio` is larger than 1, the size of `all_triples_and_labels` would be `(1+neg_ratio) * len(bp_triples)`，then the evaluations should all be divided by  `(1+neg_ratio) * len(triples)`. Looking forward to your reply.

**Comment by Mehran-k** [comment]
If by `evaluations` you mean during training the model, then `(1+neg_ratio) * len(triples)` is actually just a constant and removing it does not affect the minimization of the loss function. In fact, it gets absorbed into the `learning rate`.

But if by `evaluations` you mean during testing the model, then I should mention that during testing I do not create negative (corrupted) examples. For each triple <h, r, t> in the test set, I test the model on how well it predicts t given h and r (and also how well it predicts h given t and r).

Hope this answers your question, but let me know if anything is still not clear.

**Comment by cleverer123** [comment]
Thanks for your reply. You explanation realize me that I have confused the training part and the testing part. It's pretty clear. Appreciate!

**Comment by Mehran-k** [comment]
My pleasure! I'll close this issue then.
